extends CharacterBody2D

var bullet_velocity: Vector2 = Vector2.ZERO
@export var damage: int = 1
var shooter: Node = null   # referencia al que dispar칩

func _physics_process(delta):
	velocity = bullet_velocity
	var collision = move_and_collide(velocity * delta)

	if collision:
		var collider = collision.get_collider()

		# Debug para saber con qu칠 choca
		print("Bala colision칩 con: ", collider.name)

		# Ignorar al due침o
		if collider == shooter:
			print("Colisi칩n ignorada con el due침o: ", shooter.name)
			return

		# Si puede recibir da침o
		if collider and collider.has_method("take_damage"):
			print("Le hago da침o a: ", collider.name)
			collider.take_damage(damage)

		# Se destruye tras colisi칩n v치lida
		print("Bala destruida tras colisi칩n con: ", collider.name)
		queue_free()

# 游댳 Este m칠todo lo dispara autom치ticamente el VisibleOnScreenNotifier2D
func _on_visible_on_screen_notifier_2d_screen_exited():
	print("Bala realmente sali칩 de la pantalla, se destruye")
	queue_free()
